pagopa/io-app
# chore: [IOBP-1048] Delete transaction when interrupting a payment

## Short description
This PR addresses the deletion of transaction payments when a user interrupts the payment process by closing the in-app browser. This implementation ensures that users who interrupt a payment and attempt to pay again within a few minutes can do so without encountering the "ongoing payment" error screen.

## List of changes proposed in this pull request
- Added the handler inside the `useWalletPaymentAuthorizationModal.tsx` to delete the current transaction when the user closes the in-app browser;
- Deleted the reducer handling of the `paymentsDeleteTransactionAction.failure` action: We don't need to show anything if the transaction deletion has failed, just navigate to the correct screen.

## How to test
- Run the dev-server;
- Start a payment from the app until you reach the in-app browser content;
- Close the in-app browser by tapping the "Cancel" button on the top-left of the screen;
- Check that the app makes an HTTP request that deletes the current transaction;
